src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector.kt

```diff
@@ -453,6 +468,31 @@ private fun ImportContext.isUsefulTraitImport(usePath: String): Boolean {
         || element.canBeAccessedByTraitName
 }
 
+// `foo::bar` excludes `foo::bar` and `foo::bar::baz`, but not `foo::bar2`
```

Comment:
Hmm, why? I'd not expect such behavior. Maybe use an explicit syntax like `foo::bar::*` to activate such behavior of it is really needed?

And btw this behavior is not documented. I see only "Use path as it will appear in an import" message